common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -48,39 +49,96 @@
   [concept-type fields]
   (map #(q2e/query-field->elastic-field (keyword %) concept-type) fields))
 
+(defn- query->execution-params
+  "Returns the execution parameters extracted from the query"
+  [query]
+  (let [{:keys [page-size offset concept-type aggregations highlights scroll scroll-id]} query
+        scroll-timeout (when scroll (es-config/elastic-scroll-timeout))
+        search-type (if scroll
+                        (es-config/elastic-scroll-search-type)
+                        "query_then_fetch")
+        sort-params (q2e/query->sort-params query)
+        fields (query-fields->elastic-fields
+                 concept-type
+                 (or (:result-fields query) (concept-type+result-format->fields concept-type query)))]
+    {:version true
+     :sort sort-params
+     :size page-size
+     :from offset
+     :fields fields
+     :aggs aggregations
+     :scroll scroll-timeout
```

Comment:
:scroll is the underlying ES param. I have updated the docstring. The one exception is the :scroll-id param, which gets stripped out by my code just before it makes the call to elastisch. If this param is not nil it calls the elastisch `scroll` function and passes it in. If it is nil it calls the normal `search` function.